.github/workflows/CI.yml

```
@@ -274,3 +274,32 @@ jobs:
         run: |
           SET CHERE_INVOKING=1
           C:\cygwin64\bin\bash -l -c "bash dev/ci.sh"
+
+  slack-notification-on-failure:
+    name: Send slack notification on CI failure
+    needs:
+      - test-unix
+      - test-cygwin
+    if: ${{ failure() && github.event_name == 'push' }}
+    # Don't run this twice on PRs for branches pushed to the same repository
+    #if: ${{ !(github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get branch name
+        id: get-branch
+        run: echo ::set-output name=branch::${GITHUB_REF#refs/*/}
+      - name: Determine whether CI status changed
+        uses: Jimdo/should-i-notify-action@0f48901c482c651cba9dc19a03efc068b402789b
+        id: should_notify
+        with:
+          branch: ${{ steps.vars.outputs.branch }}
+          needs_context: ${{ toJson(needs) }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Send slack notification
+        uses: act10ns/slack@e4e71685b9b239384b0f676a63c32367f59c2522
+        if: steps.should_notify.outputs.should_send_message == 'yes'
+        with:
+          status: Failure
```

It shouldn't always be failure, the status might also be success (if the failing branch has been fixed). Judging by your helpful Markdown file, it should instead look like:
```suggestion
          status: ${{ job.status }}
```

---

Hmm, good point. Although resolving that might be a bit more complicated. IIRC, if I put `job.status` there, then the status of the `slack-notification-on-failure` job is used, which always is success. I'll have to look into that later.

---

Cool, yeah - no rush. I'm not sure what the permitted values of `status` are; if they can be anything, then maybe something like this could be made to work?
```suggestion
          status: ${{ failure() }}
```

Otherwise maybe something like this could work?
```
(Step to determine whether to post a message)

# Step to determine which message to send
- name:
  run: |
    if ${{ failure() }} then
      # set env variable STATUS
    elif ${{ success() }}
      # set env variable STATUS to be success
    fi

# Step to post the message
- uses: (whatever) 
  ...
  with:
    status: ${{ env.STATUS }}
```

---

Yes, we have to use `needs.<job>.result`, then it works! :) See 2b9c913e9ab78ee6ef

---

Cool 🙂 So according to https://github.com/ssiccha/gap/blob/ss/minimal-slack-test/.github/workflows/test-slack.yml it should be:
```suggestion
          status: ${{ steps.get-status.outputs.status }}
```

---

Oh, yes I forgot to push that ^^